#! /bin/sh /usr/share/dpatch/dpatch-run
## 65_mouse_buttons.dpatch by Mike Hommey <glandium@debian.org>
##
## All lines beginning with `## DP:' are a description of the patch.
## DP: Extended mouse buttons support taken from #244305. Thanks Peter Colberg

@DPATCH@

=== widget/src/gtk2/nsWindow.cpp
==================================================================
--- xulrunner/widget/src/gtk2/nsWindow.cpp	(revision 72)
+++ xulrunner/widget/src/gtk2/nsWindow.cpp	(local)
@@ -1543,6 +1543,9 @@
         return;
 
     switch (aEvent->button) {
+    case 1:
+        eventType = NS_MOUSE_LEFT_BUTTON_DOWN;
+        break;
     case 2:
         eventType = NS_MOUSE_MIDDLE_BUTTON_DOWN;
         break;
@@ -1550,7 +1553,7 @@
         eventType = NS_MOUSE_RIGHT_BUTTON_DOWN;
         break;
     default:
-        eventType = NS_MOUSE_LEFT_BUTTON_DOWN;
+        return;
         break;
     }
 
